source/ProcessManager.Orchestrations.Tests/Unit/Processes/BRS_021/ForwardMeteredData/V1/Thingies/MeteringPointMasterDataProviderTests.cs

```diff
@@ -0,0 +1,490 @@
+    }
+
+    [Fact]
+    public async Task Given_TwoMasterDataWithTwoEnergySuppliers_When_GetAndConvertMasterData_Then_ListOfFour()
+    {
+        _electricityMarketViews.MeteringPointMasterDataProvider = () => GetMeteringPointMasterData();
+
+        var meteringPointMasterData = await _sut.GetAndConvertMasterData(
+            "custom-metering-point-master-data",
+            "2021-01-01T00:00:00Z",
+            "2021-01-01T00:00:00Z");
+
+        meteringPointMasterData
+            .Should()
+            .HaveCount(4);
+
+        meteringPointMasterData.Should()
+            .SatisfyRespectively(
+                first =>
+                {
+                    first.MeteringPointId.Value.Should().Be("custom-metering-point-master-data");
+                    first.ValidFrom.Should().Be(new DateTimeOffset(2021, 1, 1, 0, 0, 0, TimeSpan.Zero));
+                    first.ValidTo.Should().Be(new DateTimeOffset(2021, 2, 1, 0, 0, 0, TimeSpan.Zero));
+                    first.EnergySupplier.Value.Should().Be("1111111111111");
+                },
+                second =>
+                {
+                    second.MeteringPointId.Value.Should().Be("custom-metering-point-master-data");
+                    second.ValidFrom.Should().Be(new DateTimeOffset(2021, 2, 1, 0, 0, 0, TimeSpan.Zero));
+                    second.ValidTo.Should().Be(new DateTimeOffset(2021, 3, 1, 0, 0, 0, TimeSpan.Zero));
+                    second.EnergySupplier.Value.Should().Be("2222222222222");
+                },
+                third =>
+                {
+                    third.MeteringPointId.Value.Should().Be("custom-metering-point-master-data");
+                    third.ValidFrom.Should().Be(new DateTimeOffset(2021, 3, 1, 0, 0, 0, TimeSpan.Zero));
+                    third.ValidTo.Should().Be(new DateTimeOffset(2021, 4, 1, 0, 0, 0, TimeSpan.Zero));
+                    third.EnergySupplier.Value.Should().Be("1111111111111");
+                },
+                fourth =>
+                {
+                    fourth.MeteringPointId.Value.Should().Be("custom-metering-point-master-data");
+                    fourth.ValidFrom.Should().Be(new DateTimeOffset(2021, 4, 1, 0, 0, 0, TimeSpan.Zero));
+                    fourth.ValidTo.Should().Be(new DateTimeOffset(2021, 5, 1, 0, 0, 0, TimeSpan.Zero));
+                    fourth.EnergySupplier.Value.Should().Be("3333333333333");
+                });
+    }
+
+    [Fact]
+    public async Task Given_TwoMasterDataWithOneFaultyEnergySuppliers_When_GetAndConvertMasterData_Then_Error()
+    {
+        var act = async () => await _sut.GetAndConvertMasterData(
+            "faulty-two-master-data-please",
+            "2021-01-01T00:00:00Z",
+            "2021-01-01T00:00:00Z");
+
+        await act.Should().ThrowAsync<Exception>().WithMessage("Metering point master data is not consistent");
+    }
+
+    [Fact]
+    public async Task Given_MasterDataWithEnergySupplierHole_When_GetAndConvertMasterData_Then_Error()
```

Comment:
Maybe something like this:

```suggestion
    public async Task Given_MasterDataWithEnergySupplierAndMissingMasterData_When_GetAndConvertMasterData_Then_Error()
```